cirq/optimizers/synchronize_terminal_measurements.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""An optimization pass to put as many measurements possible at the end."""
+
+from cirq import circuits, ops, protocols
+
+
+class SynchronizeTerminalMeasurements():
+    """Move measurements to the end.
+
+    Move all terminal measurements in a circuit to the final moment if it
+    can accomodate them (without overlapping with other operations). If
+    self._measure_only_moment is true then a new moment will be added to
+    the end of the circuit containing all the measurements that could be
+    brough forward.
+    """
+
+    def __init__(self, measurements_only_moment: bool = True):
```

Comment:
Rename this parameter to `after_other_operations`.

Reply:
done